### PR TITLE
Add grunt release:[releasetype] task

### PR DESCRIPTION
**Pull request by SimonWoolf**
Usage: `grunt release:patch` to increment the patch level, compile everything, commit the generated files, and tag that commit with the release version.

Can also use `minor`, `major`, `prepatch`, `prerelease` etc. instead of `patch`.

### CODE REVIEW

**Comment by paddybyers** [comment]
LGTM but how has this been tested?

**Comment by SimonWoolf** [comment]
By me trying it locally - it doesn't push anything, so easy enough to try out and revert (git reset --hard @^, git clean -f, and delete the new tag)

File: package.json [hunk 1]
```diff
@@ -24,17 +26,16 @@
     "grunt-shell": "~1.1",
     "grunt-zip": "~0.16",
     "karma": "~0.12",
-    "karma-cli": "~0.0",
     "karma-chrome-launcher": "~0.1",
-    "karma-firefox-launcher": "~0.1",
```
Comment (mattheworiordan) [inline package.json]: Why is this removed?

File: package.json [hunk 2]
```diff
@@ -24,17 +26,16 @@
...
+    "karma-cli": "~0.0",
     "karma-env-preprocessor": "~0.1",
+    "karma-firefox-launcher": "~0.1",
+    "karma-nodeunit": "~0.2",
     "karma-phantomjs-launcher": "~0.1",
     "karma-requirejs": "~0.2",
     "karma-sauce-launcher": "~0.2",
     "karma-story-reporter": "~0.3",
-    "karma-nodeunit": "~0.2",
```
Comment (mattheworiordan) [inline package.json]: And this? along with CLI

File: package.json (hunk 1 above)
Comment (SimonWoolf) [inline package.json]: It wasn't. It was moved down to line 32, karma-nodeunit was moved from 35 to 33, karma-cli from 27 to 30. npm sorts the array into alphabetical order automatically. Presumably it had got out of alphabetical order due to dependencies being added to the array manually

Comment (mattheworiordan) [inline package.json]: Ah yes, sorry, reviewed in haste

**Comment by mattheworiordan** [comment]
:+1:
